sources/packages/api/src/services/application/application.service.ts

```diff
@@ -361,4 +361,40 @@ export class ApplicationService extends RecordDataModelService<Application> {
       .addOrderBy("application.applicationNumber")
       .getMany();
   }
+
+  /**
+   * get applications of a institution location
+   * with COE status required and completed.
+   * @param locationId location id .
+   * @returns student Application list.
+   */
+  async getCOEApplications(locationId: number): Promise<Application[]> {
+    return this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationNumber",
+        "application.id",
+        "application.coeStatus",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "student",
+      ])
+      .leftJoin("application.offering", "offering")
+      .leftJoin("application.student", "student")
```

Comment:
This could be changed to a "InnerJoin". The relation application->student->user must exists.